pyomo/solvers/plugins/solvers/ASL.py

```diff
@@ -158,13 +158,15 @@ def create_command_line(self, executable, problem_files):
         # Pyomo/Pyomo) with any user-specified external function
         # libraries
         #
-        if 'PYOMO_AMPLFUNC' in env:
-            if 'AMPLFUNC' in env:
-                env['AMPLFUNC'] += "\n" + env['PYOMO_AMPLFUNC']
+        if "PYOMO_AMPLFUNC" in env:
+            if "AMPLFUNC" in env:
+                for line in env["PYOMO_AMPLFUNC"].split("\n"):
+                    if line not in env["AMPLFUNC"]:
+                        env["AMPLFUNC"] += "\n" + line
             else:
```

Comment:
There is an edge case where this would match a substring in AMPLFUNC.  Not terribly likely, as the `.so` / `.dll` extension would end up appearing in the middle of the string, but still possible.  I think a better solution would be:
```python
exitsing = set(env['AMPLFUNC'].split("\n")
for line in env["PYOMO_AMPLFUNC"].split("\n"):
    if line not in existing:
        env["AMPLFUNC"] += "\n" + line
```